### ISSUE TITLE
Malware scan fails to find EICAR string in image over 25Mb

### ISSUE THREAD

**Issue by evansgp** [anchore/anchore-engine#615]
**Version of Anchore Engine and Anchore CLI if applicable**:

```
anchore-cli --version
anchore-cli, version 0.8.1
kubectl describe pod anchore-demo-anchore-engine-analyzer-6c47f8766f-wrx6s | grep Image
    Image:         docker.io/anchore/anchore-engine:v0.8.1
    Image ID:      docker.io/anchore/anchore-engine@sha256:43e0a7fd25483c7b6d8889d892ac353d4e3f137a6c681b871269b67b2d4b5ec2
```

**What happened**:

I created a container to trial out various features of Anchore, it looks like:

```
FROM node:13

COPY credentials .
COPY package.json .
COPY index.js .
COPY node_modules/ node_modules
COPY virus .
```

Where `virus` is:

```
X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*
```

When I add the image, wait for it to complete and look at the malware content it gives:

```
anchore-cli image content 012345678910.dkr.ecr.ap-southeast-2.amazonaws.com/gareth-test:virus4 malware
Error: There is no image content (malware) to provide for 012345678910.dkr.ecr.ap-southeast-2.amazonaws.com/gareth-test:virus4
```

I turned up the logging to DEBUG to ensure it was actually scanning and it is indeed:

```
[service:worker] 2020-09-10 05:44:38+0000 [-] [Thread-168] [anchore_engine.clients.localanchore_standalone/run_anchore_analyzers()] [DEBUG] command succeeded: cmd=/usr/local/lib/python3.6/site-packages/anchore_engine/analyzers/modules/50_malware_scan.py /anchore_service f1960ece9712b2f3f6da633e051ba8f27ff11dc74ada9303d22f2563eb1880df /analysis_scratch/9bfee9d5-245c-4635-850e-58b40c57f6ea /analysis_scratch/9bfee9d5-245c-4635-850e-58b40c57f6ea/output /analysis_scratch/9bfee9d5-245c-4635-850e-58b40c57f6ea stdout=Beginning malware analyzer execution
[service:worker]        Malware scanner clamav enabled. Starting run
[service:worker]        running clamav scan
[service:worker]        Refreshing ClamAV DB
[service:worker]        Running command freshclam --stdout --datadir=/home/anchore/clamav/db --config-file=/home/anchore/clamav/freshclam.conf
[service:worker]        ClamAV scan metadata: {'db_version': {'daily': '25925', 'main': '59', 'bytecode': '331'}, 'db_update_enabled': True}
[service:worker]        Running command clamscan --suppress-ok-results --infected --recursive --allmatch --archive-verbose --tempdir=/analysis_scratch/9bfee9d5-245c-4635-850e-58b40c57f6ea --database=/home/anchore/clamav/db /analysis_scratch/9bfee9d5-245c-4635-850e-58b40c57f6ea/squashed.tar
[service:worker]        Raw ClamAV stdout: 
[service:worker]        ----------- SCAN SUMMARY -----------
[service:worker]        Known viruses: 8890276
[service:worker]        Engine version: 0.102.4
[service:worker]        Scanned directories: 0
[service:worker]        Scanned files: 1
[service:worker]        Infected files: 0
[service:worker]        Data scanned: 0.00 MB
[service:worker]        Data read: 967.80 MB (ratio 0.00:1)
[service:worker]        Time: 24.545 sec (0 m 24 s)
[service:worker]        ClamAV findings: []
[service:worker]        Analyzer output: {'malware': {'clamav': '{"scanner": "clamav", "findings": [], "metadata": {"db_version": {"daily": "25925", "main": "59", "bytecode": "331"}, "db_update_enabled": true}}'}}
[service:worker]        Handling output key malware stderr=
[service:worker] 2020-09-10 05:44:38+0000 [-] [Thread-168] [anchore_engine.clients.localanchore_standalone/run_anchore_analyzers()] [DEBUG] timing: specific analyzer time: /usr/local/lib/python3.6/site-packages/anchore_engine/analyzers/modules/50_malware_scan.py - 49.1749062538147
[service:worker] 2020-09-10 05:44:40+0000 [-] [Thread-168] [anchore_engine.clients.localanchore_standalone/analyze_image()] [DEBUG] timing: total analyzer time: 012345678910.dkr.ecr.ap-southeast-2.amazonaws.com/gareth-test@sha256:32b51c3024cb9e54bff58bfae413c005bb9b1d5ab92124ef98a02251b6abe70c - 196.2890224456787
```

During further troubleshooting I snagged the `squashed.tar` that it processes, loaded it into a container locally and ran the commands taken from the debug logs that correlated with what I saw in the code:

```
docker run -it -v $(pwd):/tmp/squased.tar docker.io/anchore/anchore-engine:v0.8.1 bash
freshclam --stdout --datadir=/home/anchore/clamav/db --config-file=/home/anchore/clamav/freshclam.conf
clamscan --suppress-ok-results --infected --recursive --allmatch --archive-verbose --tempdir=/analysis_scratch/9bfee9d5-245c-4635-850e-58b40c57f6ea --database=/home/anchore/clamav/db /tmp/squased.tar/squashed.tar 

----------- SCAN SUMMARY -----------
Known viruses: 8890276
Engine version: 0.102.4
Scanned directories: 0
Scanned files: 1
Infected files: 0
Data scanned: 0.00 MB
Data read: 967.80 MB (ratio 0.00:1)
Time: 16.878 sec (0 m 16 s)
```

It also did not find the EICAR string. *However*, when I run it on the untar'ed archive, it does find it:

```
clamscan --suppress-ok-results --infected --recursive --allmatch --archive-verbose  --database=/home/anchore/clamav/db /tmp/squased.tar/squashed/
/tmp/squased.tar/squashed/virus: Win.Test.EICAR_HDB-1 FOUND
/tmp/squased.tar/squashed/virus!(0): Win.Test.EICAR_HDB-1 FOUND

----------- SCAN SUMMARY -----------
Known viruses: 8890276
Engine version: 0.102.4
Scanned directories: 4045
Scanned files: 31645
Infected files: 1
Data scanned: 1362.94 MB
Data read: 854.34 MB (ratio 1.60:1)
Time: 204.778 sec (3 m 24 s)
```

When executing `clamscan` with `-v --debug` this shows up:

```
LibClamAV debug: cli_updatelimits: scansize exceeded (initial: 104857600, consumed: 0, needed: 1014814720)
LibClamAV debug: cli_updatelimits: filesize exceeded (allowed: 26214400, needed: 1014814720)
```

I believe the command needs to be expanded to include `--max-filesize` and `--max-scansize`, otherwise the maximum container size it will scan is 25Mb (or alternatively it could untar the file first), from the docs:

```
--max-filesize=#n
    Extract and scan at most #n kilobytes from each archive. You may pass the value in megabytes in format xM or xm, where x is a number. This option protects your system against DoS attacks (default: 25 MB, max: <4 GB) 
--max-scansize=#n
    Extract and scan at most #n kilobytes from each scanned file. You may pass the value in megabytes in format xM or xm, where x is a number. This option protects your system against DoS attacks (default: 100 MB, max: <4 GB) 
```

Unfortunately when I try to do this clamav throws errors:

```
clamscan --suppress-ok-results --infected --scan-archive=yes --recursive --allmatch --archive-verbose  --database=/home/anchore/clamav/db --max-filesize=4095m --max-scansize=4095M /tmp/squased.tar/squashed.tar
LibClamAV Warning: Bytecode run timed out in interpreter after 5000 opcodes
LibClamAV Warning: Bytecode 79 failed to run: CL_ETIMEOUT: Time limit reached
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory
LibClamAV Error: cli_malloc(): Attempt to allocate 725456912 bytes. Please report to https://bugzilla.clamav.net
LibClamAV Warning: Couldn't grow the blob: we may be low on memory

----------- SCAN SUMMARY -----------
Known viruses: 8890276
Engine version: 0.102.4
Scanned directories: 0
Scanned files: 1
Infected files: 0
Data scanned: 1301.51 MB
Data read: 967.80 MB (ratio 1.34:1)
Time: 332.930 sec (5 m 32 s)
```

I have approximately 10Gb of free memory when I run this. My `squashed.tar` is just under 1Gb, which doesn't seem to be uncommon (given this is a trival node application in an official node base image).

I think there might be two options here:
1) untar it first
2) provide the max size options

I feel like the first is the better option but I know very little about Anchore and clamav, so I will leave this with you. :)

**What did you expect to happen**:

For EICAR string to be found

**Any relevant log output from /var/log/anchore**:

Above.

**What docker images are you using**:

Above

**How to reproduce the issue**:

Above.

**Comment by zhill** [comment]
First, thanks @evansgp for a great write-up. I'm looking into this now.

**Comment by evansgp** [comment]
Thanks for looking into it @zhill ! Let me know if you need more info.

**Comment by dakaneye** [comment]
@evansgp, I linked a potential fix for this issue. I discussed with @zhill and we think that it would be easier to allow configuration of the actual **clamav** commands we use than to deal with untarring (which in itself can be a PITA). The linked proposal will allow clamscan or freshclam args to be added via the **analyzer_config.yaml**, so we can override the default values used by those commands if necessary.

**Comment by zhill** [comment]
@dakaneye can this be closed as complete?

**Comment by zhill** [comment]
Closed by #625

**Comment by nketia** [comment]
@evansgp 
Please how did you embed the EICAR signature into the image file? I am trying to have an image with embedded malware to test with Clamav
